build_ql_masters

```diff
@@ -0,0 +1,202 @@
+                if len(found_files) == 0:
+                    print ("Could not generate all NIRES masters.")
+                    sys.exit(1)
+
+
+            source_files += found_files
+            for file in found_files:
+                new_base = os.path.basename(file).replace('_2_','_1_').replace('_7_','_1_')
+                dest_files.append(os.path.join(nires_dest_dir, new_base))
+
+    if pargs.instrument == 'all' or pargs.instrument == 'keck_mosfire':
+
+        if pargs.setup == 'all' or pargs.setup == "Y_long":
+            dest_dir = os.path.join(ql_masters, 'MOSFIRE_MASTERS', 'Y')
+            os.makedirs(dest_dir, exist_ok=True)
+
+            files = ['Masters/MasterSlits_A_15_01.fits.gz',
+                     'Masters/MasterTilts_A_4_01.fits', 
+                     'Masters/MasterWaveCalib_A_4_01.fits',
+                     'Science/spec1d_m191118_0064-GD71_MOSFIRE_2019Nov18T104704.507.fits',
+                     'sens_m191118_0064-GD71_MOSFIRE_2019Nov18T104704.507.fits' ]
+
+            source_filters = [os.path.join(redux_dir, 'keck_mosfire', 'Y_long', x) for x in files]
+            
+            
+            if not pargs.force_build:
+                found_files = find_source_files(source_filters)
+            else:
+                found_files = []
+
+            if len(found_files) == 0:
+                print("Could not find all MOSFIRE masters, attempting to generate them.")
+                run_devsuite("keck_mosfire", 'Y_long', redux_dir)
+
+                found_files = find_source_files(source_filters)
+                if len(found_files) == 0:
+                    print("Failed to generate MOSFIRE masters.")
+                    sys.exit(1)
+
+            source_files += found_files
+
+            for file in found_files:
+                dest_files.append(os.path.join(ql_masters, 'MOSFIRE_MASTERS', 'Y', os.path.basename(file)))
+
+    # Copy the files
+    for source_file, dest_file in zip(source_files, dest_files):
+        copy_me(source_file, dest_file, pargs.force_copy)
+
+    exit(0)
+
+def find_source_files(source_filters):
+    """Expand any wildcards to see if any exist"""
+    found_files = []
+    for filter in source_filters:
+        found_files += glob.glob(filter)
+
+    return found_files
+
+def copy_me(new_file, cooked_file, force_copy):
+    """
+    Simple script to copy a given file to a new file
+    First compares that the new_file is newer than the
+    cooked file (if the latter exists)
+
+    Args:
+        new_file: str
+        cooked_file: str
+
+    Returns:
+
+    """
+    # Compare date stamp
+    doit = True
+    if os.path.exists(cooked_file):
+        # Time is in seconds total (like MJD)
+        if os.path.getctime(cooked_file) > os.path.getctime(new_file):
+            doit = False
+    if doit or force_copy:
+        shutil.copy2(new_file, cooked_file)
+        print("Generated/over-wrote {:s}".format(cooked_file))
+
+
+def run_devsuite(instrument, setup, redux_dir):
```

Comment:
See my point above. Does this run afterburner as well? That is needed for MOSFIRE, but I don't see that here. 

In general we may want sensfuncs to be part of quicklook, but not always which means sometimes running reductions plus afterburner (pypeit_Sensfunc) and other times now.

Reply:
It passes "develop" as the tests argument to the build_test_setup function, which causes it to add all test types like  (i.e. it adds prep, reduce, afterburn, and ql tests).  This is actually a bit of a hack because I have to tell it to *not* run quicklook tests as those would call build_ql_masters again.

I do want to rework the arguments to pypeit_test to make it easy to specify multiple instruments and setups and exactly which tests to run for each.  So it would be easy to re-run the failed tests from an overnight run.  This script could definitely benefit from that, but I didn't want to tackle that work in this PR